ladybug/datatype/base.py

```diff
@@ -46,6 +46,13 @@ def __init__(self, name=None):
         """
         self._name = name
 
+    def __key(self):
+        return (
+            self._name, self._units, self._si_units, self._ip_units, self._min,
+            self._max, self._abbreviation, self._unit_descr, self._point_in_time,
+            self._cumulative, self._normalized_type, self._type_enumeration
```

Comment:
You definitely don't want to check that the `_type_enumeration` is equal for the same reason you don't want to check that the data collection enumeration isn't equal.